test/test_distributions.py

```diff
@@ -358,6 +358,14 @@ def get_sp_dist(jax_dist):
     T(dist.Gamma, np.array([0.5, 1.3]), np.array([[1.0], [3.0]])),
     T(dist.GaussianRandomWalk, 0.1, 10),
     T(dist.GaussianRandomWalk, np.array([0.1, 0.3, 0.25]), 10),
+    T(
+        dist.GaussianCopulaBeta,
+        np.array([7.0, 2.0]),
+        np.array([4.0, 10.0]),
+        np.array([[1.0, 0.75], [0.75, 1.0]]),
+    ),
+    T(dist.GaussianCopulaBeta, 2.0, 1.0, np.eye(3)),
+    T(dist.GaussianCopulaBeta, 2.0, 1.0, np.full((5, 3, 3), np.eye(3))),
```

Comment:
Could you help me add dtype conversion logic to [those utilities](https://github.com/pyro-ppl/numpyro/blob/41669121410102245e61df473ecbe5ff8109065d/numpyro/distributions/util.py#L423)?
```
dtype = jnp.result_type(float)
jnp.asarray(a/b/x/y) -> jnp.asarray(a/b/x/y, dtype=dtype)
```